via/views/api/youtube.py

```diff
@@ -1,1328 +1,42 @@
-from pyramid.httpexceptions import HTTPNotFound
+import logging
+
+from h_pyramid_sentry import report_exception as report_exception_to_sentry
 from pyramid.view import view_config
 
+from via.services import YouTubeService
+
+logger = logging.getLogger(__name__)
+
 
 @view_config(route_name="api.youtube.transcript", permission="api", renderer="json")
 def transcript(request):
     """Return the transcript of a given YouTube video."""
 
     video_id = request.matchdict["video_id"]
 
-    if video_id != "1":
-        raise HTTPNotFound()
+    try:
+        transcript_ = request.find_service(YouTubeService).get_transcript(video_id)
+    except Exception as exc:  # pylint: disable=broad-exception-caught
+        report_exception_to_sentry(exc)
+        logger.exception(exc)
```

Comment:
The youtube-transcript-api doesn't document what types of exceptions it might raise. Regardless, I think it's actually appropriate for us to catch all types of exception here.

It'll all get reported to both Papertrail and Sentry. This may result in some noise in Papertrail and Sentry because if anything requests `/api/youtube/transcript/foo` with a non-valid YouTube video ID then it'll trigger an error response and an exception to get sent to Papertrail and Sentry. But I think it's probably best to catch too many exceptions in monitoring than too few, we want to know what's going on with our API calls to YouTube if they're failing.

Reply:
Totally agree about having a catch all at the end. For example, if you send it real nonsense (like an int or None) it raises assertion errors. Presumably other things + coding errors would also result in unpredictable things. 

It doesn't have it in the docs, but the exceptions are [exported in the main namespace](https://github.com/jdepoix/youtube-transcript-api/blob/master/youtube_transcript_api/__init__.py). The hierarchy is also pretty flat:

 * https://github.com/jdepoix/youtube-transcript-api/blob/master/youtube_transcript_api/_errors.py
 
Everything "handled" is a child of `CouldNotRetrieveTranscript`, although there are lots of sub-cases in there which we might, or might not care. So we could choose to supress all of those and log other things.

[The exported errors](https://github.com/jdepoix/youtube-transcript-api/blob/master/youtube_transcript_api/__init__.py) are:

```python
TranscriptsDisabled,
NoTranscriptFound,
CouldNotRetrieveTranscript,
VideoUnavailable,
TooManyRequests,
NotTranslatable,
TranslationLanguageNotAvailable,
NoTranscriptAvailable,
CookiePathInvalid,
CookiesInvalid,
FailedToCreateConsentCookie,
YouTubeRequestFailed,
InvalidVideoId,
```
I think being in the main name space is as close as you're going to get to an indication these are for public consumption.

Looking at the above, we might want to pay special attention to `TooManyRequests` as this is the one which is likely to get us in trouble.

Reply:
We discussed this in Slack but I think I'd like to handle this in the Sentry web UI. For example we can use that UI to filter out exceptions like `InvalidVideoId` that we don't want to be alerted for (but still have it alert us if we suddenly start getting a *lot* of these exceptions), and we can set alarms for exceptions like `TooManyRequests` that we want to get more noisily alerted for. So I think we don't need any special handling of different types of `youtube-transcript-api` exceptions in the code itself